### PR TITLE
Maxim peripherals VDDIO configuration

### PR DESCRIPTION
**Pull request by CiprianRegus**
This has to be merged first: https://github.com/analogdevicesinc/no-OS/pull/1561

Draft until properly tested on multiple MCUs

### CODE REVIEW

File: drivers/platform/maxim/max32660/maxim_spi.c [hunk 1]
```diff
@@ -151,6 +157,9 @@ static int _max_spi_config(struct no_os_spi_desc *desc)
 		goto err_init;
 	}
 
+	spi_pins.vssel = eparam->vssel;
```
Comment (buha) [inline drivers/platform/maxim/max32660/maxim_spi.c:160]: Just a remark, for the sake of completeness, if vssel is not initialized by driver API user, its value will default to 0 which means VDDIO (not VDDIOH). I think this is good behavior.

**Comment by buha** [comment]
let's not forget about documentation, perhaps you could edit these pages:
https://wiki.analog.com/resources/no-os/drivers/i2c
https://wiki.analog.com/resources/no-os/drivers/spi
https://wiki.analog.com/resources/no-os/drivers/uart

And add a collapsible area by platform where you show how exactly to fill the xuip/slave_xip on maxim.

File: drivers/platform/maxim/max32650/maxim_spi.c [hunk 2]
```diff
@@ -142,6 +151,8 @@ static int32_t _max_spi_enable_ss(uint32_t id, uint32_t chip_select)
 		return -EINVAL;
 	}
 
+	spi_pins.vssel = st->init_param->vssel;
```
Comment (amiclaus) [inline drivers/platform/maxim/max32650/maxim_spi.c:154]: i think we should do also `cs.vssel = st->init_param->vssel`. Or is the vssel for the chip select treated somewhere else?

**Comment by CiprianRegus** [comment]
Set the vssel value to CS pins.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**